audio_service/android/src/main/java/com/ryanheise/audioservice/AudioService.java

```diff
@@ -1082,7 +1068,15 @@ public void onSetShuffleMode(int shuffleMode) {
         @Override
         public void onCustomAction(String action, Bundle extras) {
             if (listener == null) return;
-            listener.onCustomAction(action, extras);
+            if (CUSTOM_ACTION_STOP.equals(action)) {
+                listener.onStop();
+            } else if (CUSTOM_ACTION_FAST_FORWARD.equals(action)) {
+                listener.onFastForward();
```

Comment:
I'm unsure now whether want to handle these backwards compatible actions here in Java or over in Dart in the `CustomActionHandler` mixin. It looks like you're doing this in two places.

Reply:
The problem is that the Dart code doesn't easily know whether or not to use custom actions. On one hand, all new code can use custom actions and not worry about compatibility issues. However, existing code using deprecated actions, without being changed, will no longer work due to Android 13 changes. That's why I added back the java code to ensure existing code keeps working no matter the Android version. And, to be safe in the future, there are now custom actions to do whatever you want.

Reply:
I can see the Java code is being used now, but doesn't that mean that the mixin can be deleted?

Reply:
The mixin pairs with the corresponding custom actions. If they are being used, the mixin can be used.  When using the original actions (non-custom) for `stop`, `fast forward`, and `rewind`, on either Android 13+ or prior, the mixin isn't needed.  The java code makes them work on Android 13+ and at the same keeps things working on older Android versions.

Reply:
I'll have to look at it again, but I think I was suspicious of whether this was stale code since the action codes for these actions were not being referenced anywhere in either the plugin or the examples.

Reply:
I can see that. I think what's confusing is that from the Dart code either native or custom controls can now be created. And to keep some backward compatibility, deprecated native controls created on the Dart side are handled as custom controls in Java.

Reply:
I've had another look at it, and I'm fairly sure this logic should not be duplicated in two places. Actually when I looked at it this time I noticed that the logic had been duplicated in 3 places (once in native, and twice in Dart).

I'll branch and make some changes to your code to refine this aspect of it and then share it with you for feedback.